src/app/backend/replicasetcommon.go

```diff
@@ -100,3 +114,78 @@ func getReplicaSetPodInfo(replicaSet *api.ReplicationController, pods []api.Pod)
 
 	return result
 }
+
+// Returns Pod Events of type warning from given list of pods
+// TODO(floreks) : use some set to get only unique error messages
+func GetPodsEventWarnings(client *client.Client, pods []api.Pod) (result []PodEvent, err error) {
+	for _, pod := range pods {
+		if !isRunningOrSucceeded(pod) {
+			log.Printf("Getting warning events from pod: %s", pod.Name)
+			events, err := GetPodEvents(client, pod)
+
+			if err != nil {
+				return nil, err
+			}
+
+			result = getPodsEventWarnings(events)
+		}
+	}
+
+	return removeDuplicates(result), nil
+}
+
+// Filters kubernetes API event objects based on event type. Empty string will return all events
+func FilterEventsByType(events []api.Event, eventType string) []api.Event {
+	if len(eventType) == 0 {
+		return events
+	}
+
+	result := make([]api.Event, 0)
+	for _, event := range events {
+		if event.Type == eventType {
+			result = append(result, event)
+		}
+	}
+
+	return result
+}
+
+// Returns list of Pod Event model objects based on kubernetes API event list object
+// Event list object is filtered to get only events of type warning.
+func getPodsEventWarnings(events *api.EventList) []PodEvent {
+	result := make([]PodEvent, 0)
+
+	for _, event := range FilterEventsByType(events.Items, api.EventTypeWarning) {
+		result = append(result, PodEvent{
+			Message: event.Message,
+			Reason:  event.Reason,
+		})
+	}
+
+	return result
+}
+
+// Removes duplicate strings from the slice
+func removeDuplicates(slice []PodEvent) []PodEvent {
+	visited := make(map[string]bool, 0)
+	result := make([]PodEvent, 0)
+
+	for _, elem := range slice {
+		if !visited[elem.Reason] {
+			visited[elem.Reason] = true
+			result = append(result, elem)
+		}
+	}
+
+	return result
+}
+
+// Returns true if given pod is in state running or succeeded, false otherwise
+func isRunningOrSucceeded(pod api.Pod) bool {
+	switch pod.Status.Phase {
+	case api.PodFailed, api.PodPending, api.PodUnknown:
```

Comment:
Hmm.... I'd reverse this statement. The reason is that if new status is added, the function will be broken. If you list running and success here, then everything would be fine.